src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveRegionDetailsFunctionalTest.java

```diff
@@ -24,38 +26,128 @@
+        responseVerificationForAll(response);
+    }
+
+    @Test
+    @ToggleEnable(mapKey = mapKey, withFeature = true)
+    @SuppressWarnings("unchecked")
+    public void getRegionDetailsByIdWithStatusCode_200() {
+        List<LrdRegionResponse> response = (List<LrdRegionResponse>)
+            lrdApiClient.retrieveRegionInfoByRegionId(HttpStatus.OK, "2");
+
+        assertThat(response).isNotNull();
+        responseVerification(response, 1);
+    }
+
+    @Test
+    @ToggleEnable(mapKey = mapKey, withFeature = true)
+    @SuppressWarnings("unchecked")
+    public void getRegionDetailsByIdsWithStatusCode_200() {
+        List<LrdRegionResponse> response = (List<LrdRegionResponse>)
+            lrdApiClient.retrieveRegionInfoByRegionId(HttpStatus.OK, "2,3");
+
+        assertThat(response).isNotNull();
+        responseVerification(response, 2);
+    }
+
+    @Test
+    @ToggleEnable(mapKey = mapKey, withFeature = true)
+    @SuppressWarnings("unchecked")
+    public void getRegionDetailsByIdAllWithStatusCode_200() {
+        List<LrdRegionResponse> response = (List<LrdRegionResponse>)
+            lrdApiClient.retrieveRegionInfoByRegionId(HttpStatus.OK, "ALL");
 
         assertThat(response).isNotNull();
-        responseVerification(response);
+        responseVerificationForAll(response);
     }
 
     @Test
     @ToggleEnable(mapKey = mapKey, withFeature = false)
     public void should_retrieve_403_when_Api_toggled_off() {
         String exceptionMessage = getFeatureFlagName().concat(" ").concat(FORBIDDEN_EXCEPTION_LD);
         validateErrorResponse(
-            (ErrorResponse) lrdApiClient.retrieveRegionInfo(HttpStatus.FORBIDDEN, ""),
+            (ErrorResponse) lrdApiClient.retrieveRegionInfoByRegionDescription(HttpStatus.FORBIDDEN, ""),
             exceptionMessage, exceptionMessage
         );
     }
 
-    private void responseVerification(LrdRegionResponse response) {
-        assertThat(response.getRegionId()).isEqualTo("2");
-        assertThat(response.getDescription()).isEqualTo("London");
-        assertThat(response.getWelshDescription()).isNull();
+    private void responseVerification(List<LrdRegionResponse> response, int expectedRegions) {
+        assertThat(response.size()).isEqualTo(expectedRegions);
+        assertThat(response.get(0).getRegionId()).isEqualTo("2");
+        assertThat(response.get(0).getDescription()).isEqualTo("London");
+        assertThat(response.get(0).getWelshDescription()).isNull();
+        if (expectedRegions == 2) {
+            assertThat(response.get(1).getRegionId()).isEqualTo("3");
+            assertThat(response.get(1).getDescription()).isEqualTo("Midlands");
+            assertThat(response.get(1).getWelshDescription()).isNull();
+        }
     }
 
+    private void responseVerificationForAll(List<LrdRegionResponse> response) {
+        assertThat(response.size()).isEqualTo(8);
+        assertThat(response.get(0).getRegionId()).isEqualTo("2");
```

Comment:
I think it looks more cleaner if the expected response is prepared in a list and the assertj library is used like below:
```assertThat(actual).hasSize(expected.size()).hasSameElementsAs(expected);```

Reply:
Addressed here: https://github.com/hmcts/rd-location-ref-api/pull/273/commits/76f41086383e79a37dc77cd8c35072c00f941c19